fc/parsing/actions.py

```diff
@@ -956,6 +1034,9 @@ def _purge_unused_mathematics(self, output_symbols):
         # Remove them and their definitions
         for symbol in unused_symbols:
             self.model.remove_variable(symbol)
+        # Add time back in to the graph if needed
+        if time_symbol not in self.model.graph.nodes:
```

Comment:
Why would it have disappeared?

Reply:
If the model modifications make the model algebraic (i.e. no ODEs) - which is the case for the clamping protocols.

Reply:
Interesting! How does that happen? The remove_variable call above shouldn't ever remove the time variable, and I don't see any remove_variable calls in cellmlmanip itself?

Reply:
It's not that the variable is removed, just that it doesn't appear in model.graph as it isn't referenced in any equations.

Reply:
Oh right, cause the graph gets rebuilt after the manipulations. Wonder if we should fix it at cellmlmanip level that the time variable is in there?

Reply:
I wondered about that. It is of course possible for a CellML file not to have time at all! And if it didn't have any ODEs in the maths, I'm not sure how you'd tell what to consider time to be?

Reply:
Whereas here I know, because it was either in the original model or the protocol created it.